src/toil/bus.py

```diff
@@ -599,5 +621,95 @@ def __init__(self) -> None:
         """
         super().__init__()
 
+    def _set_bus_and_message_types(self, bus: MessageBus, wanted_types: List[type]) -> None:
+        """
+        Connect to the given bus and collect the given message types.
+
+        We must not have connected to anything yet.
+        """
+
+        # We need to call the two inherited connection methods
+        super()._set_bus_and_message_types(bus, wanted_types)
+        self._set_bus(bus)
 
 
+@dataclass
+class JobStatus:
+    """
+    Records the status of a job.
```

Comment:
Since this is a public thing now it could use some more documentation for what the fields mean.